src/btree/bt_random.c

```diff
@@ -398,6 +431,8 @@ __wt_random_descent(WT_SESSION_IMPL *session, WT_REF **refp, uint32_t flags)
         /* Eviction just wants any random child. */
         if (eviction) {
             descent = pindex->index[__wt_random(&session->rnd) % entries];
+            if (sample_inmem_page && __wt_ref_is_root(current))
+                __random_root_inmem_ref(session, current, &descent, &session->rnd);
```

Comment:
Merge conflict - 
In later versions we pass `rnd` to this function. Here we use `&session->rnd`.